src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/BaseHeartbeatProperties.cs

```diff
@@ -0,0 +1,174 @@
+                            hasSetValues = true;
+                            break;
+                        case "osVersion":
+                            provider.AddHeartbeatProperty(fieldName, true, this.GetRuntimeOsType(), true);
+                            hasSetValues = true;
+                            break;
+                        case "processSessionId":
+                            provider.AddHeartbeatProperty(fieldName, true, this.GetProcessSessionId(), true);
+                            hasSetValues = true;
+                            break;
+                        default:
+                            provider.AddHeartbeatProperty(fieldName, true, "UNDEFINED", true);
+                            break;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    CoreEventSource.Log.FailedToObtainDefaultHeartbeatProperty(fieldName, ex.ToString());
+                }
+            }
+
+            return Task.FromResult(hasSetValues);
+        }
+
+        /// <summary>
+        /// This will return the current running .NET framework version, based on the version of the assembly that owns
+        /// the 'Object' type. The version number returned can be used to infer other things such as .NET Core / Standard.
+        /// </summary>
+        /// <returns>a string representing the version of the current .NET framework</returns>
+        private string GetRuntimeFrameworkVer()
+        {
+#if NET45 || NET46
+            Assembly assembly = typeof(Object).GetTypeInfo().Assembly;
+
+            AssemblyFileVersionAttribute objectAssemblyFileVer =
+                        assembly.GetCustomAttributes(typeof(AssemblyFileVersionAttribute))
+                                .Cast<AssemblyFileVersionAttribute>()
+                                .FirstOrDefault();
+            return objectAssemblyFileVer != null ? objectAssemblyFileVer.Version : "undefined";
+#elif NETSTANDARD1_3
+            return System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription;
+#else
+#error Unrecognized framework
+            return "unknown";
+#endif
+        }
+
+        /// <summary>
+        /// Returns the current target framework that the assembly was built against.
+        /// </summary>
+        /// <returns>standard string representing the target framework</returns>
+        private string GetBaseSdkTargetFramework()
+        {
+#if NET45
+            return "net45";
+#elif NET46
+            return "net46";
+#elif NETSTANDARD1_3
+            return "netstandard1.3";
+#else
+#error Unrecognized framework
+            return "undefined";
+#endif
+        }
+
+        /// <summary>
+        /// Runtime information for the underlying OS, should include Linux information here as well.
+        /// </summary>
+        /// <returns>String representing the OS or 'unknown'</returns>
+        private string GetRuntimeOsType()
```

Comment:
For net45, Environment.OSVersion.Platform.ToString() returns Win32NT
For netstand1_3, values would be WINDOWS, LINUX etc,
Any way to get consistent values in net45 and netstandard targets.

Can you provide example values in the comments here as well?

Reply:
I'll have a look and see if I can tighten this up a bit.